src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        const index = Animal.alive.indexOf(animal);
+
+        Animal.alive.splice(index, 1);
```

Comment:
Instead of using `splice` to remove the animal from the `Animal.alive` array, it's preferred to use the `filter` method to remove animals with health less than or equal to 0. This aligns with the checklist requirement.